app/src/org/commcare/recovery/measures/ExecuteRecoveryMeasuresPresenter.java

```diff
@@ -256,9 +257,9 @@ private void onAsyncExecutionSuccess() {
 
     private void onAsyncExecutionFailure(String reason) {
         mLastExecutionStatus = STATUS_FAILED;
-        Logger.log(LogTypes.TYPE_MAINTENANCE, String.format(
-                "%s failed with %s for recovery measure %s", mCurrentMeasure.getType(), reason, mCurrentMeasure.getSequenceNumber()));
         if (mActivity != null) {
+            Logger.log(LogTypes.TYPE_MAINTENANCE, String.format(
```

Comment:
I am not sure how `mCurrentMeasure` is getting null though I am hoping it's because actvity is no longer alive and hence hoping moving the log under activity not null check might fix it.

Reply:
Just double chekcing, you mean how "mActivity is getting through null", right? not `mCurrentMeasure`? Or is `mCurrentMeasure` coming through null when mActivity is null?

Reply:
we are getting a crash report  because of `mCurrentMeasure` being null. Though I have not been able to repro it on my end.

Reply:
I have now reverted this change and wrapped the whole thing inside a try catch block to not crash the app in case of any errors here but log the exception and show user a failure message.